transactions/src/main/java/org/jboss/as/txn/service/TxnServices.java

```diff
@@ -65,6 +65,9 @@
 
     public static final ServiceName JBOSS_TXN_HTTP_REMOTE_TRANSACTION_SERVICE = JBOSS_TXN.append("service", "http-remote");
 
+    public static final ServiceName JBOSS_TXN_CONTEXT_XA_TERMINATOR = JBOSS_TXN.append("JBossContextXATerminator");
```

Comment:
Yes, there are in fact. I was investigating on that and I'm convinced that we need two services for XATerminator. The context one and this jboss one from Narayana.

context XATerminator is used only in JCA when working with inflow transaction. Then we can see the flow
EIS -> JCA -> wfly txn client -> Narayana

The Narayana xa terminator is used by first Narayana itself. The service passes the terminator instance to Narayana which uses it to provide that implementation of XATerminator if somebody asks Narayana directly for it. I think Narayana should offers it's own terminator and not terminator of wfly client. The second and more important consumer of the narayana terminator is wildfly txn client itself. It injects the service and pass the Narayana terminator to itself for being able to delegate the call from JCA to Narayana. Service creates jboss xa terminator in version of jta or jts and wfly txn client just consume it and delegate later (and don't care more).

Is that explanation sufficient?